ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2362,10 +2362,17 @@ private Partition loadPartitionInternal(Path loadPath, Table tbl, Map<String, St
               + ", Direct insert = " + isDirectInsert + ")");
         }
         if (newFiles != null) {
-          newFileStatuses = listFilesCreatedByQuery(loadPath, writeId, stmtId);
-          newFiles.addAll(newFileStatuses.stream()
-              .map(FileStatus::getPath)
-              .collect(Collectors.toList()));
+          if (!newFiles.isEmpty()) {
+            // We already know the file list from the direct insert manifestfile
+            FileSystem srcFs = loadPath.getFileSystem(conf);
+            newFileStatuses = new ArrayList<>();
+            for (Path filePath : newFiles) {
```

Comment:
if you already have streams here, could we use foreach?

Reply:
getFileStatus throws IOException, and I do not want to wrap ip